python/semantic_kernel/connectors/memory/postgres/constants.py

```diff
@@ -0,0 +1,17 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+DEFAULT_SCHEMA = "public"
+
+# Limitation based on pgvector documentation https://github.com/pgvector/pgvector#what-if-i-want-to-index-vectors-with-more-than-2000-dimensions
+MAX_DIMENSIONALITY = 2000
+
+# Maximum number of keys that will be processed in a single batch statement
+MAX_KEYS_PER_BATCH = 1000
```

Comment:
It's not a database limitation, but for some of the batch insert/delete statements the payload could get very large for big models. 

I second guessed myself, but after having a conversation with copilot about this (copilot's output below), I think it's worthwhile to put a limit on the number of rows per transaction. However, I'm going to modify the code to make this configurable in a setting.

<details>
<summary>Copilot output:</summary>

The grouping by `MAX_KEYS_PER_BATCH` in the `_inner_upsert` method is necessary to manage the size of the data being passed into the `executemany` call. Here are the reasons and consequences of not batching the records:

_Reasons for Batching:_
- Memory Usage: Passing a large number of rows with large data into a single `executemany` call can consume a significant amount of memory. This can lead to memory exhaustion, especially if the data is very large or if the system has limited memory resources.
- Database Performance: Databases have limits on the size of a single transaction. Sending too many rows in a single transaction can exceed these limits, causing the transaction to fail. Additionally, large transactions can lock tables for extended periods, impacting the performance of other operations.
- Network Overhead: Sending a large amount of data in a single call can lead to network congestion and increased latency. Batching the data into smaller chunks can help mitigate this issue by spreading the load over multiple smaller transactions.

_Consequences of Not Batching:_
- Memory Exhaustion: Without batching, the system may run out of memory, leading to crashes or the inability to process the data.
- Transaction Failures: Large transactions may exceed the database's transaction size limits, causing them to fail and potentially leading to data loss or corruption.
- Performance Degradation: Large transactions can cause significant performance degradation, both in terms of the time taken to complete the transaction and the impact on other operations accessing the database.
- Increased Latency: Sending large amounts of data in a single call can increase network latency, leading to slower overall performance.
</details>